tensorboard/webapp/app_routing/route_config_types.ts

```diff
@@ -47,9 +47,39 @@ export interface ConcreteRouteDef {
 }
 
 export interface RedirectionRouteDef {
-  routeKind: null;
+  routeKind?: null;
   path: string;
   redirectionPath: string;
 }
 
-export type RouteDef = ConcreteRouteDef | RedirectionRouteDef;
+export interface ProgrammticRedirectionRouteDef {
+  /**
+   * A path that should activate the route definition. A Parameter, see
+   * definition in `ConcreteRouteDef`, is only used as a wild card and its value
+   * is not used meaningfully as all raw `pathParts` are passed to the
+   * `redirector`.
+   */
+  path: string;
+  /**
+   * A function that returns, from a pathParts, a new pathParts it should
+   * redirect to.
+   */
+  redirector: (pathParts: string[]) => string[];
```

Comment:
Since we don't have a dedicated type for `PathParts` and the concept is mainly used in route_config.ts which is not exposed outside of AppRouting, could we make the comment for this field explain what a `pathPart` is?